internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,261 @@
+	deployStore, err := deploy.NewStore(sessProvider, store)
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.Use(afero.NewOsFs())
+	if err != nil {
+		return nil, err
+	}
+
+	prompter := prompt.New()
+	opts := &localRunOpts{
+		localRunVars: vars,
+
+		prompt:      prompter,
+		store:       store,
+		ws:          ws,
+		deployStore: deployStore,
+	}
+	opts.envFeaturesDescriber = func(envName string) (versionCompatibilityChecker, error) {
+		envDescriber, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
+			App:         opts.appName,
+			Env:         envName,
+			ConfigStore: opts.store,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("new environment compatibility checker: %v", err)
+		}
+		return envDescriber, nil
+	}
+	return opts, nil
+}
+
+func (o *localRunOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if _, err := o.store.GetApplication(o.appName); err != nil {
+		return fmt.Errorf("get application %s: %w", o.appName, err)
+	}
+	return nil
+}
+
+func (o *localRunOpts) Ask() error {
+	if err := o.validateOrAskEnvName(); err != nil {
+		return err
+	}
+	if err := o.validateOrAskWorkloadName(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	//TODO(varun359): Get build information from the manifest and task definition for workloads
+
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskEnvName() error {
+	if o.envName != "" {
+		return o.validateEnvName()
+	}
+	envs, err := o.getDeployedEnvironments(o.appName)
+	if err != nil {
+		return err
+	}
+	if len(envs) == 0 {
+		return fmt.Errorf("no deployed environments found in the app %s", o.appName)
+	}
+	if len(envs) == 1 {
+		log.Infof("Only one environment found, defaulting to: %s\n", color.HighlightUserInput(envs[0]))
+		o.envName = envs[0]
+		return nil
+	}
+	selectedEnvName, err := o.prompt.SelectOne("Select an environment in which you want to test", "", envs, prompt.WithFinalMessage("Environment:"))
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = selectedEnvName
+	return nil
+}
+
+func (o *localRunOpts) getDeployedEnvironments(appName string) ([]string, error) {
+	envConfig, err := o.store.ListEnvironments(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("get environments for the app %s: %w", appName, err)
+	}
+
+	var envs []string
+	for _, env := range envConfig {
+		envs = append(envs, env.Name)
+	}
+
+	var deployedEnvs []string
+	for _, env := range envs {
+		isDeployed, err := o.isEnvironmentDeployed(env)
+		if err != nil {
+			return nil, err
+		}
+		if isDeployed {
+			deployedEnvs = append(deployedEnvs, env)
+		}
+	}
+	return deployedEnvs, nil
+}
+
+func (o *localRunOpts) isEnvironmentDeployed(envName string) (bool, error) {
+	var checker versionCompatibilityChecker
+
+	envDescriber, err := o.envFeaturesDescriber(envName)
+	if err != nil {
+		return false, err
+	}
+	checker = envDescriber
```

Comment:
this can be simplified to 
```suggestion
	checker, err := o.envFeaturesDescriber(envName)
	if err != nil {
		return false, err
	}
```